lib/iris/fileformats/_nc_load_rules/helpers.py

```diff
@@ -1438,24 +1407,47 @@ def build_and_add_dimension_coordinate(
         identifier=engine.cf_var.cf_name,
     )
 
-    _ = _add_or_capture(
+    problem = _add_or_capture(
         build_func=partial(
             _build_dimension_coordinate,
             cf_coord_var,
-            destination,
             coord_name,
             coord_system,
         ),
         add_method=partial(_add_dimension_coordinate, engine, cf_coord_var),
         cf_var=cf_coord_var,
         destination=destination,
     )
+    if problem is not None:
+        coord_var_name = str(cf_coord_var.cf_name)
+        stack_notes = problem.stack_trace.__notes__
+        if stack_notes is None:
+            stack_notes = []
+        stack_notes.append(
+            f"Failed to create {coord_var_name} dimension coordinate:\n"
+            f"Gracefully creating {coord_var_name!r} auxiliary coordinate instead."
+        )
+        problem.stack_trace.__notes__ = stack_notes
+        problem.handled = True
+
+        _ = _add_or_capture(
```

Comment:
This changes the behaviour slightly (as shown by your test) as the potentially masked array of point data is no longer _"Gracefully filled"_ (line 1294ish). I think it is a good thing that masked data is retained in the AuxCoord, but maybe worth highlighting in the release notes as it might break some downstream tests?

Reply:
dc3658ae1ea1c14ec3e0a891078a28aef130af97